crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs

```diff
@@ -538,6 +538,19 @@ pub(crate) struct AccessPropertyOpts {
 
     /// `true` means that the provided [Key] is crated from a computed key.
     pub is_key_computed: bool,
+
+    // ```ts
+    // interface F {
+    //   foo: string;
+    //   bar: number;
+    // }
+
+    //  var obj11: F | string;
+    //
+    //  obj11.foo; // Error TS2339
+    //
+    // ```
+    pub disallow_property_is_any: bool,
```

Comment:
Not sure about this name

Reply:
Then, do you have a name you recommend?

Reply:
A property for such operation already exists, so I used it